### DISCUSSION TITLE
Compilation on Ubuntu 21.04

### DISCUSSION THREAD

**Discussion by audetto** [XEphem/XEphem#15] [General]
I can compile in Ubuntu 21.04 but

1) no need to set MOTIF=
2) -lXp does not exist and can be safely removed

**Comment by brandon-rhodes** [comment]
Thanks for the note! I tested myself, and can confirm that XEphem compiled, and that printing to a PostScript file still worked fine.

1. Thus, I have removed the `MOTIF=` advice from the `INSTALL` file’s Linux command. For good measure, I also removed the `libXm/linux86/` directory that `MOTIF=` used to reference.
2. I have removed `-lXp`, and have containers in the `Container/` directory that confirm that XEphem can still compile and run on both Ubuntu 14.04 and 20.04 without that library.

Thanks again for the report! I had seen similar claims on the mailing list, but only now have had time to sit down and try them out myself. Hopefully this will remove an obstacle that was preventing folks from using XEphem!

**Comment by rmathar** [comment]
I created a Debian/Ubuntu package in ```https://build.opensuse.org``` . The standard way of installation for Ubuntu is:
- add a line of the format

  ``` deb https://download.opensuse.org/repositories/home:/rjmathar/xUbuntu_20.04 ./```

   to the file ```/etc/apt/sources.list```
- update the apt database and install the package

  ``` sudo apt-get update```
   ```sudo apt-get install xephem```

- add the line
    ```XEphem.ShareDir:  /usr/share/xephem```
   either to all ```~/.xephem/XEphem``` files or to the file
     ```/etc/X11/app-defaults/XEphem```
   such that the png and other auxiliary data are  found.

**Reply by brandon-rhodes** [reply]
Thanks for working to put together a package! Is there any way that the package could itself include the correct `ShareDir` setting, maybe by delivering a ready-made `/etc/X11/app-defaults/XEphem` file that already includes the default setting? (Or is the file generated in a way that the package builder doesn't let you control?)

**Reply by rmathar** [reply]
I'll add the construction of the app-defaults/XEphem file to the Makefile.am file so (hopefully) that step does not need to be done by the installer... Seems to work on a native Ubuntu 20.04
